jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -620,13 +621,27 @@ static JavaContainerBuilder getJavaContainerBuilderWithBaseImage(
 
     if (projectProperties.getMajorJavaVersion() >= 9
         || rawConfiguration.getExpandClasspathDependencies()) {
-      List<String> dependencies =
-          projectProperties
-              .getDependencies()
+      List<Path> jars = projectProperties.getDependencies();
+      List<String> duplicates =
+          jars.stream()
+              .map(path -> path.getFileName().toString())
+              .collect(Collectors.groupingBy(filename -> filename, Collectors.counting()))
+              .entrySet()
```

Comment:
nit: Maybe we could simplify the logic by put this under a separate variable (`occurrencesMap`)?